src/rules/react-usememo-should-be-component.ts

```
@@ -29,6 +29,539 @@ const isJsxElement = (node: TSESTree.Node | null): boolean => {
+    if (
+      searchNode.type === AST_NODE_TYPES.VariableDeclarator &&
+      searchNode.id.type === AST_NODE_TYPES.Identifier &&
+      searchNode.init
+    ) {
+      const targetVar = searchNode.id.name;
+      const sourceVars = extractVariableNames(searchNode.init);
+      if (!variableFlows.has(targetVar)) {
+        variableFlows.set(targetVar, new Set());
+      }
+      sourceVars.forEach((sourceVar) => {
+        variableFlows.get(targetVar)!.add(sourceVar);
```

Biome (1.9.4)

[error] Forbidden non-null assertion.

Unsafe fix: Replace with optional chain operator ?. This operator includes runtime checks, so it is safer than the compile-only non-null assertion operator


(lint/style/noNonNullAssertion)